src/widget/container.rs

```diff
@@ -0,0 +1,67 @@
+use super::Widget;
+use embedded_graphics::{
+    prelude::*,
+    primitives::{CornerRadii, PrimitiveStyleBuilder, Rectangle, RoundedRectangle},
+};
+
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub struct Border<Color> {
+    pub color: Color,
+    pub width: u32,
+}
+
+#[derive(Clone, Copy, Default, Eq, PartialEq)]
+pub struct Options<Color> {
+    pub background_color: Option<Color>,
+    pub border: Option<Border<Color>>,
+    pub corner_radii: Option<CornerRadii>,
+}
+
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub struct Container<Color> {
+    options: Options<Color>,
+}
+
+impl<Color> Container<Color>
+where
+    Color: PixelColor,
+{
+    pub fn new(options: Options<Color>) -> Self {
+        Self { options }
+    }
+}
+
+impl<Color> Widget<Color> for Container<Color>
+where
+    Color: PixelColor,
+{
+    fn draw<Display, Error>(
+        &self,
+        display: &mut Display,
+        origin: Point,
+        size: Size,
+    ) -> Result<(), Error>
+    where
+        Display: DrawTarget<Color = Color, Error = Error>,
+    {
+        let mut style = PrimitiveStyleBuilder::new();
```

Comment:
I need to get more familiar with the builder pattern (as a consumer + implementor)